### PR TITLE
Improve swap navigation

### PR DESCRIPTION
**Pull request by antonijzelinskij**
#8692xnytw
#86933fbch

### CODE REVIEW

**Comment by stepanLav** [comment]
Task linked: [CU-86933fbch Android. Improve swap navigation](https://app.clickup.com/t/86933fbch)

**Comment by stepanLav** [comment]
Task linked: [CU-8692xnytw Android. Determine swap support based on chain.json](https://app.clickup.com/t/8692xnytw)

File: app/src/main/java/io/novafoundation/nova/app/root/navigation/Navigator.kt [hunk 1]
```diff
@@ -350,9 +351,15 @@ class Navigator(
         navController?.navigate(R.id.action_mainFragment_to_swapFlow, AssetSwapFlowFragment.getBundle(payload))
     }
 
-    override fun openSwapSettings(assetPayload: AssetPayload) {
-        val payload = SwapSettingsPayload(assetPayload)
-        navController?.navigate(R.id.action_swapFlowFragment_to_swapSettings, SwapMainSettingsFragment.getBundle(payload))
+    override fun openSwapSetupAmount(assetPayload: AssetPayload) {
+        val returnTo = when (navController?.currentDestination?.id) {
+            R.id.swapFlowFragment -> SwapFinishFlowDestination.BALANCE_LIST
+            R.id.balanceDetailFragment -> SwapFinishFlowDestination.BALANCE_DETAILS
```
Comment (valentunn) [inline app/src/main/java/io/novafoundation/nova/app/root/navigation/Navigator.kt:357]: Hm I thought we wanted to always to open balance details regardless of the origin

File: feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/balance/detail/BalanceDetailFragment.kt [hunk 2]
```diff
@@ -137,6 +143,8 @@ class BalanceDetailFragment : BaseFragment<BalanceDetailViewModel>() {
 
         viewModel.sendEnabled.observe(balanceDetaiActions.send::setEnabled)
 
+        viewModel.sendEnabled.observe(balanceDetaiActions.send::setEnabled)
```
Comment (valentunn) [inline feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/balance/detail/BalanceDetailFragment.kt:146]: Seems to be a dublicated line
image

**Review by valentunn — APPROVED** [review]

File: app/src/main/java/io/novafoundation/nova/app/root/navigation/Navigator.kt (hunk 1 above)
Comment (antonijzelinskij) [reply app/src/main/java/io/novafoundation/nova/app/root/navigation/Navigator.kt]: Fixed

**Pull request merged**